### DISCUSSION TITLE
rangeStrategy `narrow`?

### DISCUSSION THREAD

**Discussion by sebthom** [renovatebot/renovate#26493] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.118.1

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### What is your question?

I'm new to renovate and I am trying to configure it for Maven projects. I have the situation where some dependencies may not be upgraded to the next major version. These dependencies have [Maven version ranges](https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges) specified as such:

```xml
<dependency>
   <groupId>org.apache.qpid</groupId>
   <artifactId>qpid-jms-client</artifactId>
    <!-- don't upgrade to 2.x which uses jakarta.jms instead of javax.jms -->
    <version>[1.10.0,2.0.0)</version>
 </dependency>
```

I know that I can avoid upgrading the major version by explicitly declaring these packages in a renovate.js but this is neither DRY and may be error prone. 
So am looking for a way where renovate would respect the current version range and only bumps the lower version limit (i.e. narrows the range) if a new version in the given range is available. E.g. in the example above from `[1.10.0,2.0.0)` to `[1.11.0,2.0.0)`.

However, it looks like none of the current [rangeStrategies](https://docs.renovatebot.com/configuration-options/#rangestrategy) is capable of doing so. These are the results I am getting:

rangeStrategy|oldValue|newValue
-|-|-
bump|`[1.10.0,2.0.0)`|`[1.10.0,2.6.0)`
replace|`[1.10.0,2.0.0)`|`[1.10.0,2.6.0)`
update-lockfile|`[1.10.0,2.0.0)`|`[1.10.0,2.6.0)`
widen|`[1.10.0,2.0.0)`|`[1.10.0,2.6.0)`
pin|`[1.10.0,2.0.0)`|`1.11.0`
in-range-only|`[1.10.0,2.0.0)`|  no new value

Is there a bug in renovate or is there any other way to achieve this via some global configuration in the config.js file?

Thanks in advance for any hints!

**Comment by rarkins** [comment]
It looks like in the above you're mixing at times a major (2.6.0) and non-major (1.10.0) update.

Renovate separates minor from major updates by default. I recommend that you configure dependency dashboard approval for all major updates, that way you don't get major updates by default but you can choose to open them on-demand.

I think that then leaves the case where you have `[1.10.0,2.0.0)`, v1.11.0 exists, and you want it to update to `[1.11.0,2.0.0)`? I think this can maybe be part of "bump" (special case when an upper bound is specified).
